cmd/agent/app/reporter/flags.go

```diff
@@ -19,39 +19,71 @@ import (
 	"fmt"
 	"os"
 	"strings"
+	"time"
 
 	"github.com/spf13/viper"
 )
 
 const (
+	prefix = "reporter."
 	// Whether to use grpc or tchannel reporter.
-	reporterType = "reporter.type"
+	reporterType = "type"
+	// Queue implementation to use
+	reporterQueueType = "queue.type"
+	// Bounded-queue size
+	boundedQueueSize = "queue.memory.size"
+	// Max retry interval
+	reporterMaxInterval = "max-retry-interval"
+	// Amount of workers for reporter
+	reporterQueueWorkers = "queue.workers"
 	// Agent tags
 	agentTags = "jaeger.tags"
 	// TCHANNEL is name of tchannel reporter.
 	TCHANNEL Type = "tchannel"
 	// GRPC is name of gRPC reporter.
 	GRPC Type = "grpc"
+
+	// MEMORY is the default reporter queue
+	MEMORY QueueType = "memory"
+
+	// DIRECT is the reporter queue for testing (no queue at all)
+	DIRECT QueueType = "direct"
```

Comment:
Could be this moved to test file?

Reply:
Referring to your comment below, we could still keep it to skip the queue (just remove the "testing only" from the description)